python/hsfs/core/vector_server.py

```diff
@@ -415,14 +416,22 @@ def get_feature_vectors(
             request_parameters is None
             or len(request_parameters) == 0
             or isinstance(request_parameters, dict)
-            or len(request_parameters) == len(entries)
-        ), "Request Parameters should be a Dictionary, None, empty or have the same length as the entries"
+            or (not entries or (entries and len(request_parameters) == len(entries)))
```

Comment:
i don't think `entries and` is necassary since `not entries` covers that case.
```suggestion
            or (not entries or (len(request_parameters) == len(entries)))
```